src/App.tsx

```diff
@@ -9,36 +9,102 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { getUsers, getPosts } from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [currentUserPosts, setCurrentUserPosts] = useState<Post[]>([]);
+  const [isFindPosts, setIsFindPosts] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [activePost, setActivePost] = useState<null | Post>(null);
+
+  const getUserPosts = async (user: User) => {
+    setIsError(false);
+    setIsFindPosts(true);
+
+    try {
+      const response = await getPosts(user.id);
+
+      setCurrentUserPosts(response);
+
+      setIsFindPosts(false);
+    } catch {
+      setIsFindPosts(false);
+      setIsError(true);
+    }
+
+    setIsFindPosts(false);
+  };
+
+  useEffect(() => {
+    getUsers()
+      .then(value => {
+        setUsers(value);
+      });
+  }, []);
+
+  // useMemo(() => setActivePost(null), [activePost]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  currentUser={currentUser as User}
+                  setCurrentUser={user => setCurrentUser(user)}
```

Comment:
```suggestion
                  onUserSelect={handleUserSelect}
```
do it in this way (naming) :arrow_up: 
and write the `handleUserSelect` function above